indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -149,6 +151,10 @@ public String getClasspathPrefix()
   @Override
   public TaskStatus run(TaskToolbox toolbox) throws Exception
   {
+    //Setting up UsedSegmentLister which is used by DatasourcePathSpec to get the list of used
+    //segments
+    DatasourcePathSpec.setUsedSegmentLister(new OverlordActionBasedUsedSegmentLister(toolbox));
```

Comment:
I don't think this can go here. If my memory is correct, this will cause hadoop classes to be loaded via DatasourcePathSpec without being in the isolated classloader in invokeForeignLoader because they are used in method return values or parameters.